src/main/java/com/minecolonies/coremod/client/gui/townhall/WindowSettingsPage.java

```diff
@@ -206,15 +205,21 @@ public void onOpened()
      */
     private void toggleHiring(@NotNull final Button button)
     {
+        String key = button.getTextAsString();
+        if (button.getText() instanceof TranslationTextComponent)
+        {
+            key = ((TranslationTextComponent) button.getText()).getKey();
```

Comment:
There is a method `getTextAsString()` which is used above it, the problem is we're comparing translated text which isn't always the best idea, so in case the component is a translation text component we have access to a translation key and arguments.
I use this key to compare to a translation constants key in order to determine which action to take, this mainly stems from the fact I do not have a `LanguageHandler.format` equivalent method to translate a string directly outside of a translation component.